main.c

```diff
@@ -395,37 +396,58 @@ void move_player_to_right(void) {
     TMR2IE = 1;
 }
 
+void seed_updt(byte val,byte refresh) {
 
-void interrupt isr(void) {
-    if (TMR2IF) {
-        TMR2IF = 0;
-        INHIBIT_PIN = 1;
-        DISPLAY_ROW_PORT = row;
-        if (display[row][col] != OFF ) {
-            DISPLAY_COLUMNS_PORT =  ((row & 0b00001000) |  col);
-            INHIBIT_PIN = 0;
-        }
-        col =  (col + 1) &  (DISPLAY_COLUMNS - 1);
-        if (col == 0) {
-            row = (row + 1) & (DISPLAY_ROWS - 1);
-        }
-    } else if (INT0F) { // left button
-        INT0F = 0;
-        key = KEY_LEFT;
-
-    } else if (INT1F) { // right button
-        INT1F = 0;
-        key = KEY_RIGHT;
-    } else if (INT2F) {
-        INT2F = 0;
-        key = KEY_ROTATE;
-    }
+  seed[seed_idx] = val;
```

Comment:
Nice seed provider. Since the TM2 timer has been used as part of the seed, I think this will generate visually a sufficient random behavior for block generation.

The unique problem I saw is the first game after the microcontroller starts. If I undestand correctly, in this moment the seed will be always the same, right? Thus the first piece of a game started from scratch will always be the same.

After the first piece we will get a pseudo-random behavior which I think is good enough.